src/ring_theory/prime.lean

```diff
@@ -0,0 +1,51 @@
+/-
+Copyright (c) 2020 Chris Hughes. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Hughes
+-/
+import algebra.associated
+import algebra.big_operators
+/-!
+# prime
+This file contains lemmas about prime elements of commutative rings.
+-/
+
+variables {R : Type*} [integral_domain R]
+open finset
+
+open_locale big_operators
+
+lemma mul_eq_mul_prime_prod {α : Type*} [decidable_eq α] {x y a : R} {s : finset α}
+  {p : α → R} (hp : ∀ i ∈ s, prime (p i)) (hx : x * y = a * ∏ i in s, p i) :
+  ∃ t u b c, t ∪ u = s ∧ disjoint t u ∧ b * c = a ∧
```

Comment:
I find this statement hard to read. Even if lean can figure out the types in one second, I had to work a bit. So I suggest adding types to `t u b c`. And maybe present the list of conclusions as a "bullet list", where the bullets are `∧`?

Reply:
Maybe also a docstring?